Cannot build on Ubuntu 16.04, 17.10, 18.04, or Debian 9.4

On all of these OSes I get build errors. For example, in 18.04 I get

```
cc -g -Wall `pkg-config --cflags libavformat libavcodec libavutil` `pkg-config --libs libavformat libavcodec libavutil` -pthread -o server segment.o buffer.o publisher.o server2.c
server2.c: In function ‘accept_thread’:
server2.c:301:66: warning: passing argument 4 of ‘av_opt_get’ from incompatible pointer type [-Wincompatible-pointer-types]
             av_opt_get(client, "method", AV_OPT_SEARCH_CHILDREN, &method);
                                                                  ^
In file included from server2.c:8:0:
/usr/include/x86_64-linux-gnu/libavutil/opt.h:737:5: note: expected ‘uint8_t ** {aka unsigned char **}’ but argument is of type ‘char **’
 int av_opt_get         (void *obj, const char *name, int search_flags, uint8_t   **out_val);
     ^~~~~~~~~~
server2.c:302:68: warning: passing argument 4 of ‘av_opt_get’ from incompatible pointer type [-Wincompatible-pointer-types]
             av_opt_get(client, "resource", AV_OPT_SEARCH_CHILDREN, &resource);
                                                                    ^
In file included from server2.c:8:0:
/usr/include/x86_64-linux-gnu/libavutil/opt.h:737:5: note: expected ‘uint8_t ** {aka unsigned char **}’ but argument is of type ‘char **’
 int av_opt_get         (void *obj, const char *name, int search_flags, uint8_t   **out_val);
     ^~~~~~~~~~
segment.o: In function `av_make_error_string':
/usr/include/x86_64-linux-gnu/libavutil/error.h:111: undefined reference to `av_strerror'
segment.o: In function `segment_free':
/root/t/mkvserver_mk2-master/segment.c:18: undefined reference to `avformat_free_context'
/root/t/mkvserver_mk2-master/segment.c:19: undefined reference to `av_free'
/root/t/mkvserver_mk2-master/segment.c:20: undefined reference to `av_free'
segment.o: In function `segment_close':
/root/t/mkvserver_mk2-master/segment.c:95: undefined reference to `av_write_trailer'
segment.o: In function `segment_init':
/root/t/mkvserver_mk2-master/segment.c:118: undefined reference to `av_malloc'
/root/t/mkvserver_mk2-master/segment.c:121: undefined reference to `avio_alloc_context'
/root/t/mkvserver_mk2-master/segment.c:123: undefined reference to `avformat_alloc_output_context2'
/root/t/mkvserver_mk2-master/segment.c:124: undefined reference to `av_opt_set_int'
/root/t/mkvserver_mk2-master/segment.c:137: undefined reference to `avcodec_alloc_context3'
/root/t/mkvserver_mk2-master/segment.c:138: undefined reference to `avcodec_parameters_to_context'
/root/t/mkvserver_mk2-master/segment.c:139: undefined reference to `avformat_new_stream'
/root/t/mkvserver_mk2-master/segment.c:141: undefined reference to `avcodec_free_context'
/root/t/mkvserver_mk2-master/segment.c:146: undefined reference to `avcodec_parameters_copy'
/root/t/mkvserver_mk2-master/segment.c:151: undefined reference to `av_dict_copy'
/root/t/mkvserver_mk2-master/segment.c:160: undefined reference to `avformat_write_header'
/root/t/mkvserver_mk2-master/segment.c:161: undefined reference to `avio_flush'
publisher.o: In function `client_disconnect':
/root/t/mkvserver_mk2-master/publisher.c:34: undefined reference to `av_write_trailer'
/root/t/mkvserver_mk2-master/publisher.c:35: undefined reference to `avio_close'
/root/t/mkvserver_mk2-master/publisher.c:36: undefined reference to `avformat_free_context'
/tmp/ccDCmLsD.o: In function `av_make_error_string':
/usr/include/x86_64-linux-gnu/libavutil/error.h:111: undefined reference to `av_strerror'
/tmp/ccDCmLsD.o: In function `read_thread':
/root/t/mkvserver_mk2-master/server2.c:64: undefined reference to `avformat_find_stream_info'
/root/t/mkvserver_mk2-master/server2.c:76: undefined reference to `avcodec_alloc_context3'
/root/t/mkvserver_mk2-master/server2.c:79: undefined reference to `avcodec_parameters_to_context'
/root/t/mkvserver_mk2-master/server2.c:94: undefined reference to `av_gettime_relative'
/root/t/mkvserver_mk2-master/server2.c:98: undefined reference to `av_read_frame'
/root/t/mkvserver_mk2-master/server2.c:109: undefined reference to `av_rescale_q'
/root/t/mkvserver_mk2-master/server2.c:110: undefined reference to `av_gettime_relative'
/root/t/mkvserver_mk2-master/server2.c:115: undefined reference to `av_gettime_relative'
/root/t/mkvserver_mk2-master/server2.c:132: undefined reference to `av_write_frame'
/root/t/mkvserver_mk2-master/server2.c:133: undefined reference to `av_packet_unref'
/root/t/mkvserver_mk2-master/server2.c:146: undefined reference to `avformat_close_input'
/tmp/ccDCmLsD.o: In function `write_segment':
/root/t/mkvserver_mk2-master/server2.c:178: undefined reference to `avformat_alloc_context'
/root/t/mkvserver_mk2-master/server2.c:184: undefined reference to `av_malloc'
/root/t/mkvserver_mk2-master/server2.c:185: undefined reference to `avio_alloc_context'
/root/t/mkvserver_mk2-master/server2.c:188: undefined reference to `avformat_open_input'
/root/t/mkvserver_mk2-master/server2.c:191: undefined reference to `av_free'
/root/t/mkvserver_mk2-master/server2.c:194: undefined reference to `avformat_find_stream_info'
/root/t/mkvserver_mk2-master/server2.c:197: undefined reference to `av_free'
/root/t/mkvserver_mk2-master/server2.c:202: undefined reference to `av_read_frame'
/root/t/mkvserver_mk2-master/server2.c:211: undefined reference to `av_write_frame'
/root/t/mkvserver_mk2-master/server2.c:212: undefined reference to `av_packet_unref'
/root/t/mkvserver_mk2-master/server2.c:215: undefined reference to `avformat_close_input'
/root/t/mkvserver_mk2-master/server2.c:216: undefined reference to `av_free'
/root/t/mkvserver_mk2-master/server2.c:222: undefined reference to `avformat_close_input'
/root/t/mkvserver_mk2-master/server2.c:223: undefined reference to `av_free'
/root/t/mkvserver_mk2-master/server2.c:224: undefined reference to `avformat_free_context'
/root/t/mkvserver_mk2-master/server2.c:225: undefined reference to `avio_context_free'
/tmp/ccDCmLsD.o: In function `accept_thread':
/root/t/mkvserver_mk2-master/server2.c:250: undefined reference to `av_dict_set'
/root/t/mkvserver_mk2-master/server2.c:255: undefined reference to `av_dict_set_int'
/root/t/mkvserver_mk2-master/server2.c:260: undefined reference to `av_dict_set_int'
/root/t/mkvserver_mk2-master/server2.c:265: undefined reference to `avio_open2'
/root/t/mkvserver_mk2-master/server2.c:279: undefined reference to `avio_accept'
/root/t/mkvserver_mk2-master/server2.c:291: undefined reference to `av_dict_set'
/root/t/mkvserver_mk2-master/server2.c:301: undefined reference to `av_opt_get'
/root/t/mkvserver_mk2-master/server2.c:302: undefined reference to `av_opt_get'
/root/t/mkvserver_mk2-master/server2.c:300: undefined reference to `avio_handshake'
/root/t/mkvserver_mk2-master/server2.c:318: undefined reference to `av_opt_set_int'
/root/t/mkvserver_mk2-master/server2.c:319: undefined reference to `av_log'
/root/t/mkvserver_mk2-master/server2.c:325: undefined reference to `avio_close'
/root/t/mkvserver_mk2-master/server2.c:330: undefined reference to `avio_close'
/root/t/mkvserver_mk2-master/server2.c:334: undefined reference to `avformat_alloc_output_context2'
/root/t/mkvserver_mk2-master/server2.c:348: undefined reference to `avcodec_alloc_context3'
/root/t/mkvserver_mk2-master/server2.c:349: undefined reference to `avcodec_parameters_to_context'
/root/t/mkvserver_mk2-master/server2.c:350: undefined reference to `avformat_new_stream'
/root/t/mkvserver_mk2-master/server2.c:351: undefined reference to `avcodec_free_context'
/root/t/mkvserver_mk2-master/server2.c:357: undefined reference to `avcodec_parameters_copy'
/root/t/mkvserver_mk2-master/server2.c:362: undefined reference to `av_dict_copy'
/root/t/mkvserver_mk2-master/server2.c:370: undefined reference to `avformat_write_header'
/root/t/mkvserver_mk2-master/server2.c:381: undefined reference to `avio_close'
/tmp/ccDCmLsD.o: In function `main':
/root/t/mkvserver_mk2-master/server2.c:440: undefined reference to `av_register_all'
/root/t/mkvserver_mk2-master/server2.c:441: undefined reference to `avformat_network_init'
/root/t/mkvserver_mk2-master/server2.c:443: undefined reference to `avformat_open_input'
collect2: error: ld returned 1 exit status
Makefile:11: recipe for target 'server' failed
make: *** [server] Error 1
```

Is there any more detail on what OS or what ffmpeg version I should be using? Readme just says ffmpeg should be "recent".

--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,8 @@ LAV_CFLAGS = `pkg-config --cflags libavformat libavcodec libavutil`
 all: server
 
 server: segment.o buffer.o publisher.o server2.c
-	$(CC) -g -Wall $(LAV_CFLAGS) $(LAV_LDFLAGS) -pthread -o server segment.o buffer.o publisher.o server2.c
+	#$(CC) -g -Wall $(LAV_CFLAGS) $(LAV_LDFLAGS) -pthread -o server segment.o buffer.o publisher.o server2.c
+	$(CC) -g -Wall $(LAV_CFLAGS) -pthread -o server segment.o buffer.o publisher.o server2.c $(LAV_LDFLAGS)
 
 segment.o: segment.c segment.h
 	$(CC) -g -Wall $(LAV_CFLAGS) -pthread -c segment.c

